src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -643,6 +643,8 @@ internal bool ContainsAnnotatedUnconstrainedTypeParameter()
 
         internal abstract TypeSymbol SetUnknownNullabilityForReferenceTypes();
 
+        internal abstract bool TryMergeNullability(TypeSymbol other, out TypeSymbol merged);
```

Comment:
>TryMergeNullability [](start = 31, length = 19)

Is the returned bool about whether a warning is required? Is it a precondition that the type and other are equivalent (i.e. have identity conversions in both directions)? I think this should require a variance parameter.